### DISCUSSION TITLE
Customisation to "plan" for a stocktake, assign it to somebody to perform, and then review it after?

### DISCUSSION THREAD

**Discussion by kriston13** [inventree/InvenTree#7508] [Q&A]
Hi!

I'm contemplating a stocktake flow where I as an admin user, can create a stocktake "event", specify that it is "for this location". Then I can go and tell my staff to go and perform "stocktake event 1123".
They'd go and use a barcode scanner + app to start scanning stock that is registered against the stocktake event.
Once they've completed it, then I can review the outcomes, and either - go ask them for clarifications - or approve it, and *then* have that stock information merged. 
This will then allow me to do things like:
- report on diffs within a location as a result of a stocktake
- have staff start a stocktake, stop to help a customer, have a different staff member take over

In my head, this is:
- new model - stocktake event
- new UIs on the web app to create & review stocktake events
- bunch of new REST APIs to create the stocktake event, add/remove parts to it, etc etc.
- the approval of it will essentially return to the existing flow where the stocktake is finalised.

Does that seem like a straightforward customisation? 

Thank you

**Comment by kriston13** [comment]
Bump? Any suggestions/advise/right-wrong-approach? First time we're touching this, so are interested in knowing if there are any pitfalls beyond what we've read on the dev doco. 
Thanks in advance!

**Reply by matmair** [reply]
This requires custom programming in form of a plugin and sufficient knowledge in python, Django and probably a bit HTML.

**Reply by matmair** [reply]
@kriston13 I would also recommend to read [this page](https://docs.inventree.org/en/stable/extend/how_to_plugin/). I have a (a bit older) sample plugin with models [here](https://github.com/invenhost/inventree-shopify).
If you find missing pieces in the plugin docs we would certainly be interested in docs issues / GitHub discussions on that.

**Comment by SchrodingersGat** [comment]
@kriston13 this would be a nice feature, perhaps you might consider it as a plugin?
